xchange-bithumb/src/main/java/org/knowm/xchange/bithumb/service/BithumbTradeService.java

```
@@ -89,12 +100,36 @@ public UserTrades getTradeHistory(TradeHistoryParams params) throws IOException
             .map(p -> ((TradeHistoryParamCurrencyPair) p).getCurrencyPair())
             .orElse(null);
     try {
-      return BithumbAdapters.adaptUserTrades(bithumbTransactions(currencyPair), currencyPair);
+      return BithumbAdapters.adaptUserTrades(
+          getBithumbUserTransactions(currencyPair).getData(), currencyPair);
     } catch (BithumbException e) {
       throw BithumbErrorAdapter.adapt(e);
     }
   }
 
+  @Override
+  public Collection<Order> getOrder(OrderQueryParams... orderQueryParams) throws IOException {
+    /* This only works for executed orders */
+    return Arrays.stream(orderQueryParams)
+        .filter(oq -> oq instanceof OrderQueryParamCurrencyPair)
+        .map(oq -> (OrderQueryParamCurrencyPair) oq)
+        .flatMap(
+            oq -> {
+              try {
+                return getBithumbOrderDetail(oq.getOrderId(), oq.getCurrencyPair()).getData()
+                    .stream()
+                    .map(detail -> BithumbAdapters.adaptOrderDetail(detail, oq.getOrderId()));
+
+              } catch (IOException e) {
+                return null;
```

silent errors are pure evil

IOException  should be thrown out